charts/k8ssandra/templates/traefik.ingressroutes.yaml

```diff
@@ -43,3 +45,92 @@ spec:
         - name: {{ $clusterName }}-{{ $datacenterName }}-service
           port: 9042
 {{- end }}
+{{- if $traefik.stargate.enabled }}
+---
+{{- if and $traefik.stargate.cassandra.enabled $traefik.cassandra.enabled }}
+  {{- fail "ingress.traefik.stargate.cassandra.enabled and ingress.traefik.cassandra.enabled cannot both be enabled." }}
+{{- end }}
+{{- $stargateHost := default "localhost" $traefik.stargate.host }}
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  name: k8ssandra-stargate
+  labels:
+{{ include "k8ssandra.labels" . | indent 4 }}
+spec:
+  rules:
+    - host: {{ $stargateHost }}
+      http:
+        paths:
+          - path: /v1/auth
+            backend:
+              serviceName: {{ $clusterName }}-{{ $datacenterName }}-stargate-service
+              servicePort: 8081
+{{- if $traefik.stargate.graphql.enabled }}
+    - host: {{ $stargateHost }}
+      http:
+        paths:
+          - path: /graphql-schema
+            backend:
+              serviceName: {{ $clusterName }}-{{ $datacenterName }}-stargate-service
+              servicePort: 8080
+          - path: /graphql/
+            backend:
+              serviceName: {{ $clusterName }}-{{ $datacenterName }}-stargate-service
+              servicePort: 8080
+  {{- if $traefik.stargate.graphql.playground.enabled }}
+          - path: /playground
+            backend:
+              serviceName: {{ $clusterName }}-{{ $datacenterName }}-stargate-service
+              servicePort: 8080
+  {{- end }}
+{{- end }}
+{{- if $traefik.stargate.rest.enabled }}
+    - host: {{ $stargateHost }}
+      http:
+        paths:
+          - path: /v2/
+            backend:
+              serviceName: {{ $clusterName }}-{{ $datacenterName }}-stargate-service
+              servicePort: 8082
+{{- end }}
+{{- if $traefik.stargate.health.enabled }}
+    - host: {{ $stargateHost }}
+      http:
+        paths:
+          - path: /checker
+            backend:
+              serviceName: {{ $clusterName }}-{{ $datacenterName }}-stargate-service
+              servicePort: 8084
+{{- end }}
+{{- if $traefik.stargate.metrics.enabled }}
```

Comment:
Do we need to expose `/metrics`? I am inclined to think no as it is primarily used internally for collecting metrics.

Reply:
Agreed, I would make sure this stays internal to the cluster. If you want external access `kubectl port-forward`

Reply:
Removed.